### DISCUSSION TITLE
Which openSSL.cnf is used?

### DISCUSSION THREAD

**Discussion by Harvey-Mushman** [openssl/openssl#21956] [Q&A] [answered]
Trying to create a self-signed certificate and after uploading my version of OpenSSL.CNF, I issue the following command:

openssl req -newkey rsa:2048 -x509 -nodes -keyout localhost.key -new -out localhost.crt -config openssl.cnf

This generates the .key and .crt files but it also adds a new file reference that is not described anywhere that I can fine.. the file is "/usr/local/ssl/openssl.cnf".  Because I did not add or point to this file, I am wondering if it is being used in the creation of my certificate?

In my browser after installing the certificate on a windows 10 box, the error I'm seeing in Chrome is as follows:  

Failed to load resource: net::ERR_CERT_AUTHORITY_INVALID

Any clues to error and/or answer to .CNF file would be appreciated. 
[openssl.cnf.txt](https://github.com/openssl/openssl/files/12515935/openssl.cnf.txt)

**Comment by kulkarniamit** [comment, marked as answer]
According to [OpenSSL Installation document](https://github.com/openssl/openssl/blob/master/INSTALL.md#openssldir), default directory for OpenSSL configuration file on Unix is `/usr/local/ssl`. The default configuration directory of your installation can also be found using  `$ openssl version -d`.

> I am wondering if it is being used in the creation of my certificate?

You can use `strace` ([man strace](https://man7.org/linux/man-pages/man1/strace.1.html)) to check the configuration file being used while generating the self-signed certificate.
```
$ strace -e trace=open,openat -o /tmp/strace.log.0 openssl req \
-newkey rsa:2048 -x509 -nodes -keyout localhost.key \
-new -out localhost.crt
...

$ grep "openssl.cnf" /tmp/strace.log.0
openat(AT_FDCWD, "/etc/pki/tls/openssl.cnf", O_RDONLY) = 3

```
